sky/skylet/providers/command_runner.py

```diff
@@ -229,6 +229,15 @@ def run_init(self, *, as_head: bool, file_mounts: Dict[str, str],
                     f'The `image_env` is:\n{image_env}')
                 raise e
 
+            # Edit docker config first to avoid the know issue in
+            # https://github.com/NVIDIA/nvidia-container-toolkit/issues/48
+            self.run(
+                'sudo jq \'.["exec-opts"] = ["native.cgroupdriver=cgroupfs"]\' '
+                '/etc/docker/daemon.json > /tmp/daemon.json;'
+                'sudo mv /tmp/daemon.json /etc/docker/daemon.json;'
+                'sudo systemctl restart docker',
+                run_env='host')
```

Comment:
Could we run some test for Azure to test out this part of the code, to make sure it works nice for initial launch, exec, stop and restart, etc?

Reply:
Sounds good! Will add some test commands in `test_job_queue_with_docker`

Reply:
@Michaelvll Relevant smoke test passed except for #3451. Please take another look 👀